packages/astro/src/runtime/server/render/component.ts

```diff
@@ -132,6 +132,9 @@ async function renderFrameworkComponent(
 			// we can skip the `check` calls and use that renderer
 			if (renderers.length === 1) {
 				renderer = renderers[0];
+
+				// Still call `check` because some renderers store state during the check phase that they expect when `renderToStaticMarkup` is called.
+				await renderer.ssr.check.call({ result }, Component, props, children)
```

Comment:
What's the point of this call? This PR was intended to remove this check to not render solid-js component twice (and actually to remove double rendering for all the frameworks)
Here, if you call it again... well... you can just remove this `if` statement

Reply:
I guess, it's possible to extend `renderer` interface with boolean flag `shouldRun` in order to explicitly say that this renderer should be called because it has some state which depends on the initial call?

You discovered that:
> some renderers (Preact in particular), expect both check and renderToStaticMarkup to be called because they store state in check
Looks like it's not really obvious from the renderer interface, so that's why I suggest to extend it.

Reply:
Hmm, it's a bit strange that function `check` (which should be pure) has some store in it

Reply:
What do you mean by "resubmit" against `next`? `next` is a function which is called inside `check`?

Reply:
@elite174 sorry for not explaining. We can't change the expectation that both `check` and `renderToStaticMarkup` both get called in 4.x. We already know it breaks out built-in renderers. That means it might also break external renderers.

We can make that a breaking change in 5.0 though, which is currently in beta. We are using the `next` branch for 5.0. If this PR was for 5.0 we could make this a breaking change, update the Preact renderer, document the change, etc.

Reply:
Oh, yes, that would be awesome.

Reply:
Just rebased the PR